.github/workflows/publish-service-app-images.yml

```diff
@@ -0,0 +1,312 @@
+name: Build Docker images and push to Dockerhub
+
+on:
+  workflow_dispatch:
+
+  push:
+    branches: [main, dev, test]
+    paths:
+      - "apps/**"
+      - "services/**"
+      - "examples/**"
+env:
+  REGISTRY: ghcr.io
+  WORKFLOW_SERVICE_IMAGE_NAME: ${{ github.repository_owner }}/workflows-service
+  WEBSOCKET_SERVICE_IMAGE_NAME: ${{ github.repository_owner }}/websocket-service
+  BACKOFFICE_APP_IMAGE_NAME: ${{ github.repository_owner }}/backoffice-app
+  WORKFLOW_DASHBOARD_APP_IMAGE_NAME: ${{ github.repository_owner }}/workflows-dashboard-app
+  HEADLESS_EXAMPLE_IMAGE_NAME: ${{ github.repository_owner }}/headless-example
+  DEV_IMAGE_TAG: dev
+  PROD_IMAGE_TAG: latest
+  TEST_IMAGE_TAG: test
+
+jobs:
+  build-workflow-services:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install jq
+        run: sudo apt-get install jq
+
+      - name: Get package version from package.json
+        id: get_version
+        run: |
+          PACKAGE_VERSION=$(jq -r '.version' services/workflows-service/package.json)
+          echo "::set-output name=version::$PACKAGE_VERSION"
+
+      - name: Login to Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
```

Comment:
Todo: change them to github container registry credentials
Similarly for the rest of the apps and services